### PR TITLE
chore: grammar support for phase independent specifier

### PR DESCRIPTION
**Pull request by Chriscbr**
First step towards implementing the `unphased` functions RFC (https://github.com/winglang/wing/pull/1711)

## Checklist

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingc/src/parser.rs [hunk 1]
```diff
@@ -1271,13 +1271,23 @@ impl<'s> Parser<'s> {
...
+
+		let class_phase = if let Some(phase) = self.get_phase_specifier(&class_modifiers)? {
+			phase
 		} else {
-			class_phase
+			scope_phase
 		};
+		if class_phase == Phase::Independent {
+			self
+				.with_error::<Node>("Unphased classes are not yet supported", &statement_node)
```
Comment (hasanaburayyan) [inline libs/wingc/src/parser.rs:1287]: Can we use the `UNIMPLEMENTED_GRAMMARS` map for this? Or maybe add a tracking issue like we do for other not yet implemented.

File: libs/wingc/src/parser.rs [hunk 2]
```diff
@@ -1706,24 +1761,33 @@ impl<'s> Parser<'s> {
...
-		let phase = match self.get_modifier("inflight_specifier", &modifiers)? {
-			Some(_) => Phase::Inflight,
-			None => phase,
+		let phase = match self.get_phase_specifier(&modifiers)? {
+			Some(phase) => phase,
+			None => scope_phase,
 		};
+		if phase == Phase::Independent {
+			self
+				.with_error::<Node>("Unphased functions are not yet supported", &func_def_node)
```
Comment (hasanaburayyan) [inline libs/wingc/src/parser.rs:1788]: Same as previous comment

File: libs/wingc/src/type_check.rs [hunk 3]
```diff
@@ -5332,6 +5332,8 @@ new cloud.Function(@inflight("./handler.ts"), lifts: { bucket: ["put"] });
 	/// * `phase` - initializer phase
 	///
 	fn check_class_field_initialization(&mut self, scope: &Scope, fields: &[ClassField], phase: Phase) {
+		// Traverse the AST of the constructor (preflight or inflight) to find all initialized fields
+		// that were initialized during its execution.
```
Comment (hasanaburayyan) [inline libs/wingc/src/type_check.rs:5336]: Like the comments 👍

**Review by hasanaburayyan — APPROVED** [review]
LGTM

File: libs/wingc/src/parser.rs (hunk 1 above)
Comment (Chriscbr) [reply libs/wingc/src/parser.rs:1287]: Sure thing, I'll add some issue links

**Pull request merged**